Mathlib/Order/Partition/Finpartition.lean

```diff
@@ -472,31 +479,69 @@ theorem existsUnique_mem (ha : a ∈ s) : ∃! t, t ∈ P.parts ∧ a ∈ t := b
 /-- The part of the finpartition that `a` lies in. -/
 def part (a : α) : Finset α := if ha : a ∈ s then choose (hp := P.existsUnique_mem ha) else ∅
 
-lemma part_mem (ha : a ∈ s) : P.part a ∈ P.parts := by simp [part, ha, choose_mem]
+@[simp]
+lemma part_mem : P.part a ∈ P.parts ↔ a ∈ s := by
+  by_cases ha : a ∈ s <;> simp [part, ha, choose_mem]
 
-lemma mem_part (ha : a ∈ s) : a ∈ P.part a := by
-  simp [part, ha, choose_property (p := fun s => a ∈ s) P.parts (P.existsUnique_mem ha)]
+@[simp]
+lemma part_eq_empty : P.part a = ∅ ↔ a ∉ s :=
+  ⟨fun h has ↦ P.ne_empty (P.part_mem.2 has) h, fun h ↦ by simp [part, h]⟩
 
-lemma part_eq_of_mem (ht : t ∈ P.parts) (hat : a ∈ t) : P.part a = t := by
-  apply P.eq_of_mem_parts (P.part_mem _) ht (P.mem_part _) hat <;> exact mem_of_subset (P.le ht) hat
+@[simp]
+lemma part_nonempty : (P.part a).Nonempty ↔ a ∈ s := by
+  simpa only [nonempty_iff_ne_empty] using P.part_eq_empty.not_left
+
+@[simp]
+lemma part_subset (a : α) : P.part a ⊆ s := by
+  by_cases ha : a ∈ s
+  · exact P.le <| P.part_mem.2 ha
+  · simp [P.part_eq_empty.2 ha]
+
+@[simp]
+lemma mem_part_self : a ∈ P.part a ↔ a ∈ s := by
+  by_cases ha : a ∈ s
+  · simp [part, ha, choose_property (p := fun s => a ∈ s) P.parts (P.existsUnique_mem ha)]
+  · simp [P.part_eq_empty.2, ha]
+
+@[deprecated (since := "2025-02-07")]
```

Comment:
What do you think of keeping this lemma around instead of deprecating it? cf things like mem_insert

Reply:
OK, I'll restore it tonight or tomorrow.